src/util/irep_serialization.cpp

```diff
@@ -93,11 +93,8 @@ void irep_serializationt::read_irep(
     reference_convert(in, r);
   }
 
-  if(in.get()!=0)
-  {
-    std::cerr << "irep not terminated\n";
-    throw 0;
-  }
+  // if fails, irep is not terminated
+  POSTCONDITION(in.get()==0);
```

Comment:
No, this deserialisation code - this is a proper exception from userland and should be treated as such (also, no side-effectful calls in invariants)